doc/flakes.adoc

```diff
@@ -0,0 +1,2 @@
+[[ch-flakes]]
+== Using home-manager with Flakes
```

Comment:
Generally speaking, "Home Manager" applies for the project in general and and "home-manager" for the command line tool and when otherwise referring to the project in file paths. Many use "home-manager" in both senses and I don't mind that but I figure the text within the project should be consistent.